wicket-native-websocket/wicket-native-websocket-core/src/main/java/org/apache/wicket/protocol/ws/WebSocketSettings.java

```diff
@@ -388,9 +401,65 @@ public Integer getSecurePort()
 	public static class SameThreadExecutor implements Executor
 	{
 		@Override
-		public void run(Runnable command)
-		{
+		public void run(Runnable command) {
 			command.run();
 		}
 	}
+
+	public static class WebSocketPushMessageExecutor implements Executor
+	{
+		/**
+		 * An executor that uses the caller thread to execute the runnable
+		 */
+		private final java.util.concurrent.Executor nonHttpRequestExecutor;
+		/**
+		 * An executor that is used when the WebSocket push is initiated in
+		 * http worker thread. In this case the WebSocket processing should be
+		 * off-loaded to a different thread that should wait for the page instance
+		 * lock.
+		 */
+		private final java.util.concurrent.Executor httpRequestExecutor;
+
+		/**
+		 * For non-http worker threads pushes the WebSocket runnable in the same request.
+		 * For http worker threads uses an elastic thread pool of 1-8 threads.
+		 */
+		public WebSocketPushMessageExecutor()
+		{
+			this(Runnable::run, new ThreadPoolExecutor(1, 8,
```

Comment:
shall this be configurable?

Reply:
It is configurable by directly using the called constructor.

Reply:
my bad :(